src/include/miopen/solver.hpp

```diff
@@ -177,9 +157,78 @@ struct SolverMixin : SolverBase
     }
 };
 
-// Typedef for convolution solvers
+/// Typedef for convolution solvers
 using ConvSolver = SolverMixin<ConvolutionContext>;
 
+/// Base class for searchable solvers
+struct SearchableSolverBase : ConvSolver
+{
+    /// Initializes performance config to the default values.
+    /// The function may involve some heuristic to guess the best solution
+    /// configuration. It is assumed that the function takes constant time
+    /// to finish and does not run kernels to measure performance etc.
+    /// The function shall always return valid config.
+    virtual boost::any GetPerformanceConfig_(const ConvolutionContext& ctx) const = 0;
```

Comment:
To be honest, I don't like naming interface public methods with _. Also, with the signature change I would suggest renaming this method to something like `GetDefaultPerformanceConfig`, for the sake of readability.

Reply:
`GetPerformanceConfig()` has been renamed to `GetDefaultPerformanceConfig()`